clever/src/selfcheck.py

```diff
@@ -622,18 +622,35 @@ def check_clever_service():
     j.add_disjunction()
     j.add_match(UNIT='clever.service')
     node_errors = []
+    xml_errors = []
+    current_xml_error_count = 0
     r = re.compile(r'^(.*)\[(FATAL|ERROR)\] \[\d+.\d+\]: (.*)$')
+    xml_error_header = 'Invalid roslaunch XML syntax:'
     for event in j:
         msg = event['MESSAGE']
-        if ('Stopped Clever ROS package' in msg) or ('Started Clever ROS package' in msg):
+        if 'Started Clever ROS package' in msg:
+            current_xml_error_count = 0
+        elif 'Stopped Clever ROS service' in msg:
+            # Clear node errors (we don't care about them anymore)
             node_errors = []
+            if current_xml_error_count == 0:
+                # We didn't get any XML errors on last launch,
+                # assume last configuration was good enough
+                xml_errors = []
         elif ('[ERROR]' in msg) or ('[FATAL]' in msg):
             msg = r.search(msg).groups()[2]
             if msg in node_errors:
                 continue
             node_errors.append(msg)
+        elif xml_error_header in msg:
+            msg = msg.replace(xml_error_header, 'clever.launch syntax error:')
```

Comment:
`clever.launch` is not only launch-file we have.

Reply:
As far as I can tell, `roslaunch` only writes the offending .launch file if it was `<include>`d. Changed the string in https://github.com/CopterExpress/clever/pull/161/commits/45435f303278bb5c5c2387b98a493e7f3e2f84b1, though it's going to complain about `clever.launch` by default.